crates/cow-amm/src/amm.rs

```diff
@@ -83,8 +87,35 @@ impl Amm {
         // The helper contract returns exactly that format but in our code base we
         // expect the signature to not already include the signer address (the parts
         // will be concatenated in the encoding logic) so we discard the first 20 bytes.
-        let raw_signature = signature.0.into_iter().skip(20).collect();
-        let signature = Signature::Eip1271(raw_signature);
+        let signature = Signature::Eip1271(raw_signature.0.iter().skip(20).cloned().collect());
+
+        // Cow AMM pools can have specific requirements for the signature validity.
+        // https://sepolia.etherscan.io/address/0xaceb697457db8bb567e7d8e4411c5364ca07101e#code
+        // Check the signature only for non-legacy pools
+        if !self
+            .helper
+            .is_legacy(self.contract.address())
+            .call()
+            .await?
```

Comment:
Is there a reason to only check the non-legacy pools?
IMO this is the correct thing to do anyway and we don't really gain anything from omitting very very few RPC requests with this check.
Also I think the actual helper interface doesn't even have this function.

Reply:
cc @sunce86 for a response here.

Reply:
The issue appears to be that the legacy pools require the pre-interaction to run to make the signature valid.
This is not easy to fake. I tried using the multicall contract but specifically the old cow amms require the pre-interaction to be called inside a settlement.
I guess the correct solution would be to execute these calls inside the settlement contract context (like we do in the balances helper contract). That way everything should appear as if it actually happens for real so all assertions should pass.